app/service/database_dump.py

```
@@ -0,0 +1,118 @@
+import logging
+import re
+import subprocess
+from datetime import datetime
+from pathlib import Path
+
+from app.config import (
+    ADMIN_POSTGRES_DATABASE,
+    ADMIN_POSTGRES_HOST,
+    ADMIN_POSTGRES_PASSWORD,
+    ADMIN_POSTGRES_USER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.setLevel(logging.INFO)
+
+
+def validate_postgres_param(value: str) -> str:
+    """
+    Allow only alphanumeric, dots, underscores and hyphens in Postgres parameters.
+
+    :param value: The Postgres parameter to validate
+    returns str: The validated parameter
+    :raises ValueError: If the parameter is invalid
+    """
+    if value is None or not re.match(r"^[\w.-]+$", value):
+        raise ValueError(f"Invalid Postgres parameter: {value}")
+    return value
+
+
+def delete_local_file(file_path: str) -> None:
```

suggestion: you could make this return a `bool`, and handle whether it successfully deleted the file using a conditional in the calling function rather than this try-except block with a bare `raise` (not good practice)

---

I'm not sure I agree about returning a bool... That still requires the caller to check the return value to decide if the action worked or not, rather than assuming it has if no exception was thrown. Seems like more effort for less clarity. If a function fails to do what it was written to do, it should throw. Also, exceptions give more context of what went wrong by default.

---

both approaches are subjective! my comment is a suggestion, I'm okay with it not being actioned:))

---

will update the bare raise, but yes in agreement as I think I am leaning into the fail fast principle, handling errors upfront (and raising them lol) and ensuring it fails loudly rather than leaving it to the caller to handle the result.